src/stores/AccountController.ts

```diff
@@ -100,23 +110,29 @@ export const initializeAccount = async () => {
 };
 
 export async function updateUser(values: { firstName: string; lastName: string } | { email: string; confirmationPassword: string }) {
-  const { auth, user } = useAccountStore.getState();
+  await withAccountService(async ({ accountService, sandbox }) => {
+    useAccountStore.setState({ loading: true });
 
-  if (!auth || !user) throw new Error('no auth');
+    const { auth, user, canUpdateEmail } = useAccountStore.getState();
 
-  const { cleengSandbox } = useConfigStore.getState().getCleengData();
+    if (Object.prototype.hasOwnProperty.call(values, 'email') && !canUpdateEmail) {
+      throw new Error('Email update not supported');
+    }
 
-  const response = await cleengAccountService.updateCustomer({ ...values, id: user.id.toString() }, cleengSandbox, auth.jwt);
+    if (!auth || !user) throw new Error('no auth');
 
-  if (!response) {
-    return { errors: Array.of('Unknown error') };
-  }
+    const response = await accountService.updateCustomer({ ...values, id: user.id.toString() }, sandbox, auth.jwt);
```

Comment:
Isn't this going to end up erasing the consents in the user metadata in the inplayer service?  Consents are not passed in here and are not populated in the service.

Reply:
There is a check in the inplayer service for that not to happen:
![image](https://user-images.githubusercontent.com/37805865/206480505-9ceebe1e-3b7e-4f93-93fc-b6c4473ef970.png)

Reply:
But I thought updates to the user object were all or nothing? Is metadata treated differently?
https://github.com/jwplayer/ott-web-app/pull/199#discussion_r1039555389

Reply:
Yes, there is an exception for the metadata, even though it's a put method it does not removes the values which are not sent.

Reply:
OK, then all these changes to support setting metadata through the service.updateCustomer should be rolled back.

This means the only 'required' parameter is fullname, right? Anything else will be part of the metadata.

Reply:
consents are not always included in the `updateCustomer` call, they are included only if they are present in the values. They are only present in the values if the `updateCustomer` is called from `updateConsents`.
`...(values?.metadata?.consents && { consents: JSON.stringify(values.metadata?.consents) })`.

`updateConsents` is not making direct api call, but calls `updateCustomer`.

Reply:
My point is not that they are included in every api call, but they are added to the update customer method when they don't need to be. This is going to be really hard for other developers to understand and is likely to lead to bugs.  It also forces us to couple 2 actions that don't need to be coupled (in addition to forcing more rework in cleeng and controller code.)

Reply:
That's the coupling that shouldn't be happening.  updateCustomer and updateConsents can both call `InPlayer.Account.updateAccount`.  The common logic can even be put into a shared method, but by requiring the updateCustomer method on the account services to also accept consents, you're forcing changes onto the cleeng service that don't need to happen.

Reply:
That's another approach and it's okay, but still don't see how this approach is affecting cleeng?

Reply:
Because the services have to have the same method interfaces, this means that the update customer method on Cleeng also has to accept and handle consent data.

Reply:
Well yes, but that's the case with other methods as well, the flows are just not the same and we want to use generic types and methods. 
Cleeng service for updating customer is not handling consents, it's an optional prop, it works same as before.